### DISCUSSION TITLE
Live edit in devcontainer?

### DISCUSSION THREAD

**Discussion by Stwissel** [quarkusio/quarkus#44882] [Q&A] [answered]
I'm using Quarkus and its VSCode plugin in a [devcontainer](http://containers.dev).

I noticed, at least with resources, that an edit/safe didn't seem to trigger a live reload (I might have missed it). So I have questions:

- where can I see that `quarkus dev` actually triggered a refresh (a log?)

- What do I need to configure? `quarkus dev` runs inside the devcontainer, while the browser runs on the host. So I guess there are ports other than 8080 to be opened?

Advice is deeply appreciated

**Comment by cescoffier** [comment]
The hot reload does not happen until you send a HTTP request. Are you saying it does not happen in this case either?

**Reply by Stwissel** [reply]
So I have to hit refresh in the browser? I expected a [vite](https://vite.dev/) like experience: you hit save on a resource / source and the browser reloads. What do I miss?

**Reply by cescoffier** [reply]
We only reload on HTTP requests or Kafka messages or anything "incoming." If you use Quinoa there is a second mechanism. @ia3andy can tell you more about it.

**Reply by ia3andy** [reply, marked as answer]
There is no vite like experience using core Quarkus as is, because it doesn't make much sense to have it for just Rest or similar. 

When doing Web/Full-Stack though, [the Web Bundler extension](https://docs.quarkiverse.io/quarkus-web-bundler/dev/index.html) adds support for browser live-reload and plenty of other feature to have an awesome web experience... out of the box (See [doc](https://docs.quarkiverse.io/quarkus-web-bundler/dev/advanced-guides.html#web-live-coding)).

It's also possible using [Quinoa extension](https://quarkiverse.github.io/quarkiverse-docs/quarkus-quinoa/dev/index.html) which seamlessly allows to connect with the npm-ecosystem

**Reply by Stwissel** [reply]
awesome. Thx for chipping in. So the answer is "there's a plugin for that". 😀
